examples/platform/silabs/SiWx917/BaseApplication.cpp

```diff
@@ -394,45 +382,60 @@ void BaseApplication::ButtonHandler(AppEvent * aEvent)
     // start blinking within the FACTORY_RESET_CANCEL_WINDOW_TIMEOUT
     if (aEvent->ButtonEvent.Action == SL_SIMPLE_BUTTON_PRESSED)
     {
-        if (!mFunctionTimerActive && mFunction == kFunction_NoneSelected)
+        if ((!mFunctionTimerActive) && (mFunction == kFunction_NoneSelected))
         {
-            StartFunctionTimer(FACTORY_RESET_TRIGGER_TIMEOUT);
             mFunction = kFunction_StartBleAdv;
+            StartFunctionTimer(FACTORY_RESET_TRIGGER_TIMEOUT);
+
+            // Wait for sometime to determine button is pressed for Factory reset
+			// other functionality
+			vTaskDelay(1000);
         }
     }
-    else
+
+	while(!(RSI_NPSSGPIO_GetPin(NPSS_GPIO_0))) {
```

Comment:
I guess it technically works but this prevents the app task from doing anything else for 5 seconds. In the EFR32 platform, we set up interrupts on push and on release of the button. Can't you do the same with the siwx917 ? What calls `sl_button_on_change` in your case

Reply:
Yes Junior

But it only prevents app for doing anything if we hold the button, So If we hold the button then we are sure that we are going to trigger Factory reset.

If we leave the hold then it immediately comes out of while loop.

Unfortunately  the interrupt is only configured for push of the button and we call sl_button_on_change   on first push interrupt and this logic determines if the button is stay pressed or released.

Reply:
This issue is solved in new HW version (A01) , we were only seeing it in older HW version ( A00)